autosklearn/automl.py

```diff
@@ -300,7 +301,15 @@ def _clean_logger(self):
         # Clean up the logger
         if self.logging_server.is_alive():
             self.stop_logging_server.set()
-            self.logging_server.join()
+
+            # We try to join the process, after we sent
+            # the terminate event. Then we try a join to
+            # nicely join the event. In case something
+            # bad happens with nicely trying to kill the
+            # process, we execute a terminate to kill the
+            # process.
+            self.logging_server.join(timeout=5)
+            self.logging_server.terminate()
```

Comment:
This should do the job, but in case we want to be super-sensitive here, we could also use `Psutil.kill()`.